models/lr/__main__.py

```diff
@@ -71,6 +70,12 @@ def evaluate_split(model, vectorizer, processor, args, split='dev'):
         save_path = os.path.join(args.save_path, dataset_map[args.dataset].NAME)
         os.makedirs(save_path, exist_ok=True)
 
+    if train_examples:
+        train_features = vectorizer.fit_transform([x.text for x in train_examples])
```

Comment:
Hmm but this means that we would `fit_transform` twice right? I am not sure if you ran this on our larger datasets (IMDB or Yelp) but it's a pretty memory and compute intensive operation.

Reply:
Yes, the same fit transform would also be applied in `trainers/bow_trainer.py`. It's just tricky because you need to initialize the LR model having the vocab size and the `BagOfWordsTrainer` requires that model.

Would it be ok to pass the `train_features` to the `BagOfWordsTrainer`? It's only used by the LR model anyway. It just feels clunky since `vectorizer` is only passed to `BagOfWordsTrainer` to compute train / eval features.

Not sure whats a better way to do this.